### PR TITLE
Add detach operation to datasets and filesets

### PR DESCRIPTION
**Pull request by marlosin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our conventions](../CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```
## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/api/dataops/dataset.ts [hunk 1]
```diff
@@ -125,6 +125,27 @@ export class Dataset<
       filter,
     );
   }
+
+  /**
+   * Creates a Detach query.
+   * @param   resourceName The name of the resource to be detached.
+   * @param   filter Filtering criterion or a callback that returns one,
+   * that will be applied to the resource to be detached.
+   * @returns ActionQuery object specialized for detaching resources to the current one.
```
Comment (KutsenkoA) [inline src/api/dataops/dataset.ts]: "from the current one"?

File: src/api/fileops/fileset.ts [hunk 2]
```diff
@@ -123,6 +123,27 @@ export class Fileset<FormDataType extends IFormData<F>, T, D, F> implements IRes
     );
   }
 
+  /**
+   * Creates a Detach query.
+   * @param   resourceName The name of the resource to be detached.
+   * @param   filter Filtering criterion or a callback that returns one,
+   * that will be applied to the resource to be detached.
+   * @returns ActionQuery object specialized for detaching resources to the current one.
```
Comment (KutsenkoA) [inline src/api/fileops/fileset.ts]: the same as above

File: src/api/dataops/dataset.spec.ts [hunk 3]
```diff
@@ -79,4 +78,23 @@ describe("Dataset class", () => {
...
 
+  describe("On detach", () => {
+    it("should return the query by passing a filter", () => {
+      const dataset = new Dataset("test", createMockFor(RequestExecuter));
+      const query = dataset.detach(
+        faker.random.word(),
+        getRandomFilteringCriteria(),
+      );
+
+      expect(query instanceof ActionQuery).toBeTruthy();
```
Comment (KutsenkoA) [inline src/api/dataops/dataset.spec.ts]: I think we need a test that `detach()` returns an ActionQuery with ActionType = `detach`, don't we?

Comment (marlosin) [reply src/api/dataops/dataset.spec.ts]: done!

**Pull request merged**